alchemy/src/supabase/function.ts

```diff
@@ -0,0 +1,190 @@
+import type { Context } from "../context.ts";
+import {
+  Resource,
+  ResourceKind,
+  ResourceID,
+  ResourceFQN,
+  ResourceScope,
+  ResourceSeq,
+} from "../resource.ts";
+import { Scope } from "../scope.ts";
+import {
+  createSupabaseApi,
+  type SupabaseApiOptions,
+  type SupabaseApi,
+} from "./api.ts";
+import { handleApiError } from "./api-error.ts";
+
+export interface FunctionProps extends SupabaseApiOptions {
+  projectRef: string;
+  name?: string;
+  body?: string;
+  importMap?: Record<string, string>;
+  entrypointUrl?: string;
+  verifyJwt?: boolean;
+  adopt?: boolean;
+  delete?: boolean;
+}
+
+export interface FunctionResource extends Resource<"supabase::Function"> {
+  id: string;
+  slug: string;
+  name: string;
+  status: string;
+  version: number;
+  createdAt: string;
+  updatedAt: string;
+}
+
+export function isFunction(resource: Resource): resource is FunctionResource {
+  return resource[ResourceKind] === "supabase::Function";
+}
+
+export const Function = Resource(
```

Comment:
All resources need docs with "@example" that mirror the data samples in the .md docs.